src/platform/user/profile/vet360/components/MobilePhone.jsx

```diff
@@ -1,22 +1,36 @@
 import React from 'react';
-import environment from 'platform/utilities/environment';
+import { connect } from 'react-redux';
 import PhoneField from './PhoneField';
 import ReceiveTextMessages from '../containers/ReceiveTextMessages';
 import { FIELD_NAMES } from '../constants';
+import { profileShowReceiveTextNotifications } from '../selectors';
 
-export default function MobilePhone() {
-  return (
-    <>
-      <PhoneField
-        title="Mobile phone number"
-        fieldName={FIELD_NAMES.MOBILE_PHONE}
-      />
-      {!environment.isProduction() && (
-        <ReceiveTextMessages
+class MobilePhone extends React.Component {
+  render() {
+    return (
+      <>
+        <PhoneField
           title="Mobile phone number"
           fieldName={FIELD_NAMES.MOBILE_PHONE}
         />
-      )}
-    </>
-  );
+        {this.props.showReceiveTextNotifications && (
+          <ReceiveTextMessages
+            title="Mobile phone number"
+            fieldName={FIELD_NAMES.MOBILE_PHONE}
+          />
+        )}
+      </>
+    );
+  }
 }
+
+export function mapStateToProps(state) {
+  return {
+    showReceiveTextNotifications: profileShowReceiveTextNotifications(state),
+  };
+}
+
```

Comment:
You've got some failing unit tests. Without digging into them too deeply, you'll probably need to make a few adjustments.

1. Export the non-connected component from this file. So `export { MobilePhone };` should do it.
2. Then in any test file that is using this component, make sure it's using the non-connected component (the new named export) rather than the connected component that's now the default export from this file.

Reply:
I did the export.  I don't think any particular tests are using this (that I can find).  (But I'm having difficulty accessing jenkins to see what's going on in particular.)

Reply:
I'm guessing if you run the tests locally you'll get the same failures that Jenkins is getting.